spyder/plugins/findinfiles/widgets.py

```diff
@@ -181,40 +196,157 @@ def find_string_in_file(self, fname):
+
                 if not self.case_sensitive:
                     line = line.lower()
+
                 if self.text_re:
                     for match in re.finditer(text, line):
                         with QMutexLocker(self.mutex):
                             if self.stopped:
                                 return False
                         self.total_matches += 1
-                        self.sig_file_match.emit((osp.abspath(fname),
-                                                  lineno + 1,
-                                                  match.start(),
-                                                  match.end(), line_dec),
-                                                 self.total_matches)
+                        self.partial_results.append((osp.abspath(fname),
+                                                     lineno + 1,
+                                                     match.start(),
+                                                     match.end(),
+                                                     line_dec))
+                        if len(self.partial_results) > (2**self.power):
+                            self.process_results()
+                            if self.power < self.max_power:
+                                self.power += 1
                 else:
                     found = line.find(text)
                     while found > -1:
                         with QMutexLocker(self.mutex):
                             if self.stopped:
                                 return False
                         self.total_matches += 1
-                        self.sig_file_match.emit((osp.abspath(fname),
-                                                  lineno + 1,
-                                                  found,
-                                                  found + len(text), line_dec),
-                                                 self.total_matches)
+                        self.partial_results.append((osp.abspath(fname),
+                                                     lineno + 1,
+                                                     found,
+                                                     found + len(text),
+                                                     line_dec))
+                        if len(self.partial_results) > (2**self.power):
+                            self.process_results()
+                            if self.power < self.max_power:
+                                self.power += 1
+
                         for text, enc in self.texts:
                             found = line.find(text, found + 1)
                             if found > -1:
                                 break
+
         except IOError as xxx_todo_changeme:
             (_errno, _strerror) = xxx_todo_changeme.args
             self.error_flag = _("permission denied errors were encountered")
+
         self.completed = True
 
+    def process_results(self):

```

Comment:
Please add a docstring to this method to understand what it does.

Reply:
Done